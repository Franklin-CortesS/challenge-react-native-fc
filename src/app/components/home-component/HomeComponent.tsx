import React from "react";
import { StyleSheet, View } from 'react-native';
import { Button } from "@react-native-elements/base";

const HomeComponent = ({navigation}: any) => {
    return (
        <View data-testid="view-1" style={styles.container}>
            <Button
            data-testid="btn-1"
            buttonStyle={styles.button}
            onPress={() => navigation.push('Tasks')}
            title="Tasks">

            </Button>
            <Button
            data-testid="btn-2"
            buttonStyle={styles.button2}
            onPress={() => navigation.push('Lists')}
            title="List">
            </Button>
        </View>
    )
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        justifyContent: 'center',
        marginHorizontal: 16,
    },
    button: {
        marginBottom: 5
    },
    button2: {
        backgroundColor: 'rgba(111, 202, 186, 1)',
    }
});

export default HomeComponent;