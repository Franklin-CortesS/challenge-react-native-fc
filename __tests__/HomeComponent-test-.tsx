/**
 * @format
 */

 import 'react-native';
 import React from 'react';
 
 // Note: test renderer must be required after react-native.
 import renderer from 'react-test-renderer';
 import HomeComponent from '../src/app/components/home-component/HomeComponent';
 import { render, fireEvent, waitFor } from "@testing-library/react-native";
 
 describe('Test para Home Component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<HomeComponent />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('navigates on Tasks button press', () => {
    const push = jest.fn();
    const { getByText } = render(<HomeComponent navigation={{ push }} />);
    fireEvent.press(getByText('Tasks'));
    expect(push).toHaveBeenCalledWith('Tasks');
  });
  
  it('navigates on List button press', () => {
    const push = jest.fn();
    const { getByText } = render(<HomeComponent navigation={{ push }} />);
    fireEvent.press(getByText('List'));
    expect(push).toHaveBeenCalledWith('Lists');
  });
 })
 