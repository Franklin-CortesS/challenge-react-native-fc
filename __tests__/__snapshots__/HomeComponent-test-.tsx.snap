// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test para Home Component renders correctly 1`] = `
<View
  data-testid="view-1"
  style={
    Object {
      "flex": 1,
      "justifyContent": "center",
      "marginHorizontal": 16,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "borderRadius": 3,
          "overflow": "hidden",
        },
        undefined,
        false,
      ]
    }
    testID="RNE_BUTTON_WRAPPER"
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "busy": false,
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#2089dc",
            "borderColor": "#2089dc",
            "borderRadius": 3,
            "borderWidth": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "marginBottom": 5,
            "padding": 8,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 18,
              "paddingVertical": 1,
              "textAlign": "center",
            }
          }
        >
          Tasks
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "borderRadius": 3,
          "overflow": "hidden",
        },
        undefined,
        false,
      ]
    }
    testID="RNE_BUTTON_WRAPPER"
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "busy": false,
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(111, 202, 186, 1)",
            "borderColor": "#2089dc",
            "borderRadius": 3,
            "borderWidth": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "padding": 8,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 18,
              "paddingVertical": 1,
              "textAlign": "center",
            }
          }
        >
          List
        </Text>
      </View>
    </View>
  </View>
</View>
`;
